packages/rx_bloc_cli/mason_templates/bricks/feature_onboarding/__brick__/lib/feature_onboarding_email_confirmation/blocs/onboarding_email_confirmation_bloc.dart

```diff
@@ -0,0 +1,116 @@
+
+  /// Opens a fake email confirmation link with a success result
+  /// Used for demo purposes, should be removed in a real app
+  void openMockDeepLinkSuccess();
+
+  /// Opens a fake email confirmation link with an error result
+  /// Used for demo purposes, should be removed in a real app
+  void openMockDeepLinkError();
+}
+
+/// A contract class containing all states of the OnboardingEmailConfirmationBloC.
+abstract class OnboardingEmailConfirmationBlocStates {
+  /// The loading state
+  Stream<bool> get isLoading;
+
+  /// The error state
+  Stream<ErrorModel> get errors;
+
+  /// State which returns a list of available email apps on the device
+  Stream<List<MailApp>> get openMailApp;
+
+  /// The email state
+  Stream<String> get email;
+
+  /// The state indicating whether the "Send new link" button is active
+  Stream<bool> get isSendNewLinkActive;
+}
+
+@RxBloc()
+class OnboardingEmailConfirmationBloc extends $OnboardingEmailConfirmationBloc {
+  OnboardingEmailConfirmationBloc(
+    this._email,
+    this._openMailAppService,
+    this._onboardingService,
+    this._countDownService,
+  ) {
+    /// Used for demo purposes, should be removed in a real app
+    _$openMockDeepLinkSuccessEvent
+        .switchMap(
+            (_) => _onboardingService.openMockEmailSuccessLink().asResultStream())
+        .setResultStateHandler(this)
+        .listen((_) {})
+        .addTo(_compositeSubscription);
+
+    /// Used for demo purposes, should be removed in a real app
+    _$openMockDeepLinkErrorEvent
+        .switchMap(
+            (_) => _onboardingService.openMockEmailErrorLink().asResultStream())
+        .setResultStateHandler(this)
+        .listen((_) {})
+        .addTo(_compositeSubscription);
+  }
+
+  final String _email;
+  final OpenMailAppService _openMailAppService;
+  final OnboardingService _onboardingService;
+  final CountDownService _countDownService;
+
+  @override
+  Stream<ErrorModel> _mapToErrorsState() => errorState.mapToErrorModel();
+
+  @override
+  Stream<bool> _mapToIsLoadingState() => loadingState;
+
+  @override
+  Stream<List<MailApp>> _mapToOpenMailAppState() => _$openMailClientEvent
+      .switchMap(
+          (title) => _openMailAppService.openMailApp(title).asResultStream())
+      .setResultStateHandler(this)
+      .whereSuccess();
+
+  @override
+  Stream<String> _mapToEmailState() => _$sendNewLinkEvent
+      .switchMap((value) => _resendConfirmationEmail().asResultStream())
+      .setResultStateHandler(this)
+      .whereSuccess()
+      .startWith(_email);
+
+  static const countDownSeconds = 10;
+
+  @override
+  Stream<bool> _mapToIsSendNewLinkActiveState() =>
+      _$disableSendNewLinkEvent.switchMap((value) => _countDownService
+          .startCountDown(countDown: countDownSeconds)
+          .map((event) => event > 0));
+
+  Future<String> _resendConfirmationEmail() async {
+    final user = await _onboardingService.resendConfirmationEmail();
+    _$disableSendNewLinkEvent.add(null);
```

Comment:
To improve readability I guess you can use `sendNewLink(null)`.

Reply:
Done